client/main.cpp

```diff
@@ -122,10 +128,25 @@ int main(int argc, char** argv)
 
     parser::KlassRegister klass_register;
     parser::ProtocolReader reader(&klass_register, std::move(stream), true);
-    client::ChromeTraceListener chrome_listener(std::move(tracepoint_map));
     std::string out_file = create_output_path(output_path.c_str());
-    chrome_listener.init(out_file);
-    reader.register_events_listener([&chrome_listener] (const parser::Event& event) { chrome_listener.process_event(event); });
+
+
+    if (format == "chrome")
+    {
+        client::ChromeTraceListener chrome_listener(std::move(tracepoint_map));
+        chrome_listener.init(out_file);
+        reader.register_events_listener([&chrome_listener] (const parser::Event& event) { chrome_listener.process_event(event); });
+    }
+    else if (format == "callgrind")
+    {
+        client::CallgrindListener callgrind_listener(std::move(tracepoint_map));
```

Comment:
If this doesn't work yet, we should print some message so users know that it's not implemented yet.